### PR TITLE
add secondary filter options by tags to assertion-check-attribute

### LINKED ISSUE #1053: Add descriptions/tags to attributes
**Objective:** Have a description or tags associated to an attribute name to improve the search attribute experience. Example, you should be able to type seconds or milliseconds or ms, and see that we have a `tracetest.span.duration` attribute.

**AC1:**
As a user adding checks in the create assertion form,
I should be able to search for an attribute based on the `description` or `tags` associated with it.

**Note:**
The description or tags should be hardcoded in the frontend for now, and they should match the [OTel semantic convention.](https://github.com/open-telemetry/opentelemetry-specification/tree/main/specification/trace/semantic_conventions)

### PR DESCRIPTION
**Pull request by cescoferraro**
This PR adds secondary filter options by tags to assertion-check-attribute

## Changes

- implements #1053 

## Fixes
Fixes #1053 

## Checklist

- [x] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [x] added a test

### CODE REVIEW

File: web/src/constants/SpanAttribute.constants.ts [hunk 1]
```diff
@@ -8,10 +8,14 @@ export const TraceTestAttributes = {
...
   TRACETEST_RESPONSE_BODY: 'tracetest.response.body',
   TRACETEST_RESPONSE_HEADERS: 'tracetest.response.headers',
-  TRACETEST_SELECTED_SPANS_COUNT: 'tracetest.selected_spans.count'
+  TRACETEST_SELECTED_SPANS_COUNT: 'tracetest.selected_spans.count',
 };
 
-export const Attributes = {
+export const AttributesTags: Record<string, string[]> = {
+  TRACETEST_SPAN_DURATION: ['ms', 'time'],
+};
```
Comment (jorgeepc) [inline web/src/constants/SpanAttribute.constants.ts:16]: Looks good man. We need to create more tags and descriptions for attributes following the OTel semantic convention https://github.com/open-telemetry/opentelemetry-specification/tree/main/specification/trace/semantic_conventions

Comment (xoscar) [reply web/src/constants/SpanAttribute.constants.ts:16]: Yes, the idea is to grab each of description text and add show it as part of the attribute input in the create assertion form.

Comment (cescoferraro) [reply web/src/constants/SpanAttribute.constants.ts:16]: OK. Will add those I find on the docs. But will leave a placeholder for all the one we go so far.

File: web/src/constants/AttributesTags.ts [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+  // NET_HOST_CARRIER_MCC: {tags:[],description:""},
+  // NET_HOST_CARRIER_MNC: {tags:[],description:""},
+  // NET_HOST_CARRIER_NAME: {tags:[],description:""},
+  // NET_HOST_CONNECTION_SUBTYPE: {tags:[],description:""},
+  // NET_HOST_CONNECTION_TYPE: {tags:[],description:""},
+  // NET_HOST_IP: {tags:[],description:""},
+  // NET_HOST_NAME: {tags:[],description:""},
+  // NET_HOST_PORT: {tags:[],description:""},
+  // NET_PEER_IP: {tags:[],description:""},
+  NET_PEER_NAME: {description: 'Name of the database host.', tags: ['example.com']},
```
Comment (cescoferraro) [inline web/src/constants/AttributesTags.ts:226]: @xoscar is that what you need?

Comment (xoscar) [reply web/src/constants/AttributesTags.ts:226]: Yeah, something like this, I'd suggest having this as json files. After that we need to add the description to the attribute dropdown.

Comment (cescoferraro) [reply web/src/constants/AttributesTags.ts:226]: @xoscar where in the attribute dropdown?

File: web/src/constants/SpanAttribute.constants.ts (hunk 1 above)
Comment (kdhamric) [reply web/src/constants/SpanAttribute.constants.ts]: Can we leverage the yaml files that contain these definitions, for example: https://github.com/open-telemetry/opentelemetry-specification/blob/main/semantic_conventions/trace/http.yaml

**Comment by xoscar** [comment]
@cescoferraro I see that we have something like this with your changes:
Screen Shot 2022-08-23 at 11 01 17

I think this is quite difficult to read and doesn't add too much value. Maybe we can ask @olha23 to give us a hand around what would be the best way to show it.

**Comment by olha23** [comment]
@xoscar @cescoferraro adding the name attribute and the meaning for attribute in dropdown seams wrong for me. If you think that users do not know the meanings of attributes, we can add the link for doc page with all explanations at the top of the input

**Comment by olha23** [comment]
@kdhamric i am sure that if we add the description as a part of what is searched - this will not solve the problem, it will only confuse the user even more because it will be very overwhelmed. The other way how we can help them - maybe think about implementing smart search. User types some characters inside the search box, the smart search function will find all the best matching attributes from our inner database. And like when user type "time" - instead 'No data' message show "Maybe you mean Duration". Or something like this 😂

**Comment by cescoferraro** [comment]
@kdhamric  the PR is already searching it by three elements
- its key
- its description
- its tags.

Screen Shot 2022-08-24 at 11 47 16

The issue here is whether should we concatenate the description after the attribute key. Just like this screenshot Oscar posted before

Screen Shot 2022-08-23 at 11 01 17

**Comment by kdhamric** [comment]
Perhaps we remove the 'dash description' from the dropdown list you are showing and just show the attribute name. On hover over an attribute name, show the attribute name - description just like you are. This may clean it up a bit. I would like to see that in an environment where we can play with it (beta?) to see how it feels.

**Comment by cescoferraro** [comment]
@kdhamric  I removed the dash descriptor thing. But the tooltip just show the text written on the select option. It is something done automatically by antd. Let me try to keep the description on the tooltip. But anyhow, the PR is ready to be reviewed/approved  @xoscar @jorgeepc

**Review by xoscar — APPROVED** [review]
LGTM

**Pull request merged**